internal/controller/mem_consistency_test.go

```diff
@@ -151,14 +152,26 @@ func watchListEvents(t testutil.TestingTB, events ...*pbresource.WatchEvent) pbr
 
 	// Return the events in the specified order as soon as they are requested
 	for _, event := range events {
+		evt := event
 		watchListClient.EXPECT().
 			Recv().
 			RunAndReturn(func() (*pbresource.WatchEvent, error) {
-				return event, nil
+				return evt, nil
```

Comment:
This was actually capturing a loop variable and returning the last event N times, instead of each of N events 1 time.